src/python_testing/TC_DRLK_2_13.py

```diff
@@ -0,0 +1,663 @@
+            TestStep("29a", "TH sends SetUser Command to DUT to create an Aliro user",
+                     "DUT sends SUCCESS response"),
+            TestStep("29b",
+                     "TH performs repeated number of SetCredential commands with credentials as 8 'startcredentialindex' until 'max_aliro_keys_supported - 1', startcredentialindex initially has value 1 .",
+                     "Verify that the DUT responds with SetCredentialResponse command and Status success."),
+            TestStep("30",
+                     "TH sends SetCredential Command to DUT with CredentialType as AliroEvictableEndpointKey for the 'alirouser' ",
+                     "Verify that the DUT responds with SetCredentialResponse command and Status success. This step will fill the last slot with credentialType as AliroEvictableEndpointKey"),
+            TestStep("31",
+                     "TH sends SetCredential Command to DUT with CredentialType as AliroNonEvictableEndpointKey and number of credentials for 'alirouser' exceeds the max_aliro_keys_supported",
+                     "Verify that the DUT responds with SetCredentialResponse command and Status success."),
+            TestStep("32", "TH sends GetCredentialStatus Command with Credential as 7 startcredentialindex+1",
+                     "DUT responds with GetCredentialStatusResponse Command and CredentialExists is false"),
+            TestStep("33", "TH sends GetCredentialStatus Command with Credential as 8 startcredentialindex+1",
+                     "Verify DUT responds with GetCredentialStatusResponse having CredentialExists is true and UserIndex as 1"),
+            TestStep("34", "TH sends ClearCredential Command to DUT to clear the ALIRO CredentialType",
+                     "Verify that the DUT sends SUCCESS response"),
+            TestStep("35", "TH sends ClearUser Command to DUT with the UserIndex as 1",
+                     "Verify that the DUT sends SUCCESS response"),
+        ]
+
+        return steps
+
+    async def read_attributes_from_dut(self, endpoint, cluster, attribute, expected_status: Status = Status.Success):
+        try:
+            attribute_value = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster,
+                                                                             attribute=attribute)
+            asserts.assert_equal(expected_status, Status.Success)
+            return attribute_value
+        except Exception as e:
+            logging.error(e)
+            asserts.assert_equal(expected_status, Status.Success,
+                                 f"Error reading attributes, response={attribute_value}")
+
+    async def aliro_attribute_verifiers(self, aliro_attribute_verify_steps: list[AliroAttributeVerify]):
+
+        for aliro_attribute_step in aliro_attribute_verify_steps:
+            th_step = aliro_attribute_step.th_step
+            self.step(th_step)
```

Comment:
are we supposed to step even if condition is false. Unsure about UI needs here, maybe we should move this later as:

```py
if pics_condition:
   self.step(th_step)
else:
   self.skip_step(th_step)
```

Reply:
the self.pics_gaurd will take care of it, it will evaluate the pics condition and skip.